OsmAnd/src/net/osmand/plus/utils/CollatorFilteredAdapter.java

```diff
@@ -0,0 +1,73 @@
+package net.osmand.plus.utils;
+
+import static net.osmand.CollatorStringMatcher.StringMatcherMode.CHECK_STARTS_FROM_SPACE;
+
+import android.content.Context;
+import android.widget.ArrayAdapter;
+import android.widget.Filter;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import net.osmand.CollatorStringMatcher;
+import net.osmand.util.Algorithms;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CollatorFilteredAdapter extends ArrayAdapter<String> {
+
+	private final List<String> originalCollection;
+	private final List<String> filteredCollection;
+
+	public CollatorFilteredAdapter(@NonNull Context context, int resource, @NonNull List<String> objects) {
+		super(context, resource, objects);
+		this.originalCollection = new ArrayList<>(objects);
+		this.filteredCollection = new ArrayList<>(objects);
+	}
+
+	@Override
+	public int getCount() {
+		return filteredCollection.size();
+	}
+
+	@Nullable
+	@Override
+	public String getItem(int position) {
+		return filteredCollection.get(position);
+	}
+
+	@NonNull
+	@Override
+	public Filter getFilter() {
+		return new Filter() {
+			@Override
+			protected FilterResults performFiltering(CharSequence constraint) {
+				FilterResults filterResults = new FilterResults();
+				List<String> suggestions = new ArrayList<>();
+
+				if (Algorithms.isEmpty(constraint)) {
+					suggestions.addAll(originalCollection);
+				} else {
+					String filterPattern = constraint.toString().toLowerCase().trim();
+					CollatorStringMatcher matcher = new CollatorStringMatcher(filterPattern, CHECK_STARTS_FROM_SPACE);
```

Comment:
Several other Filters use [NameStringMatcher(namePart, CHECK_CONTAINS)](https://github.com/osmandapp/Osmand/blob/dfac98bc21d4bf713c9fe1f7ca298f3c09ac4151/OsmAnd/src/net/osmand/plus/download/local/dialogs/LocalSearchFilter.java#L43), probably here we need the same one, if not specified otherwise